okhttp/src/main/java/okhttp3/CertificatePinner.kt

```diff
@@ -305,23 +305,18 @@ class CertificatePinner internal constructor(
         publicKey.encoded.toByteString().sha256()
 
     internal fun newPin(pattern: String, pin: String): Pin {
-      val canonicalHostname = when {
-        pattern.startsWith(WILDCARD) -> {
-          "http://${pattern.substring(WILDCARD.length)}".toHttpUrl().host
-        }
-        else -> {
-          "http://$pattern".toHttpUrl().host
-        }
-      }
+      require(!pattern.endsWith("*")) { "Invalid pattern: TLD cannot be a wildcard" }
```

Comment:
This line bugs me, because lots of other patterns are invalid.  e.g. * in the middle.

We should express this something closer to, "Patterns must start with wildcard when present".